src/Conf.ml

```diff
@@ -806,36 +837,39 @@ let to_absolute file =
 let read_config ~filename conf =
   read_conf_files conf ~dir:(Filename.dirname (to_absolute filename))
 
-let config =
-  { break_cases= C.get Formatting.break_cases
-  ; break_infix= C.get Formatting.break_infix
-  ; break_string_literals= C.get Formatting.break_string_literals
-  ; break_struct= Poly.(C.get Formatting.break_struct = `Force)
-  ; comment_check= C.get comment_check
-  ; disable= C.get Formatting.disable
-  ; doc_comments= C.get Formatting.doc_comments
-  ; escape_chars= C.get Formatting.escape_chars
-  ; escape_strings= C.get Formatting.escape_strings
-  ; extension_sugar= C.get Formatting.extension_sugar
-  ; field_space= C.get Formatting.field_space
-  ; if_then_else= C.get Formatting.if_then_else
-  ; indicate_nested_or_patterns=
-      C.get Formatting.indicate_nested_or_patterns
-  ; infix_precedence= C.get Formatting.infix_precedence
-  ; leading_nested_match_parens=
-      C.get Formatting.leading_nested_match_parens
-  ; let_and= C.get Formatting.let_and
-  ; let_open= C.get Formatting.let_open
-  ; margin= C.get Formatting.margin
-  ; max_iters= C.get max_iters
-  ; module_item_spacing= C.get Formatting.module_item_spacing
-  ; ocp_indent_compat= C.get Formatting.ocp_indent_compat
-  ; parens_tuple= C.get Formatting.parens_tuple
-  ; quiet= C.get quiet
-  ; sequence_style= C.get Formatting.sequence_style
-  ; type_decl= C.get Formatting.type_decl
-  ; wrap_comments= C.get Formatting.wrap_comments
-  ; wrap_fun_args= C.get Formatting.wrap_fun_args }
+let update_using_cmdline conf =
+  let update f conf =
+    match C.get f with None -> conf | Some x -> C.update_field f conf x
+  in
+  conf
+  |> update Formatting.break_cases
+  |> update Formatting.break_infix
+  |> update Formatting.break_string_literals
+  |> update Formatting.break_struct
+  |> update comment_check |> update Formatting.disable
+  |> update Formatting.doc_comments
+  |> update Formatting.escape_chars
+  |> update Formatting.escape_strings
+  |> update Formatting.extension_sugar
+  |> update Formatting.field_space
+  |> update Formatting.if_then_else
+  |> update Formatting.indicate_nested_or_patterns
+  |> update Formatting.infix_precedence
+  |> update Formatting.leading_nested_match_parens
+  |> update Formatting.let_and
+  |> update Formatting.let_open
+  |> update Formatting.margin |> update max_iters
+  |> update Formatting.module_item_spacing
+  |> update Formatting.ocp_indent_compat
+  |> update Formatting.parens_tuple
+  |> update quiet
+  |> update Formatting.sequence_style
+  |> update Formatting.type_decl
+  |> update Formatting.wrap_comments
+  |> update Formatting.wrap_fun_args
```

Comment:
It would be nice if this could be refactored into something such as an update for the entire record, so that when a field is added to the type, we will get a compilation error here.

Reply:
Folding over the `store` defined above should do it.

Reply:
Even better